resources/idlharness.js

```diff
@@ -1828,28 +1828,28 @@ IdlInterface.prototype.test_self = function()
                 // only way to get our hands on it. We only test that its class
                 // string, at least, is correct.
                 assert_class_string(Object.getPrototypeOf(this.get_interface_object().prototype),
-                                    inherit_interface + 'Prototype',
+                                    inherit_interface,
                                     'Class name for prototype of ' + this.name +
-                                    '.prototype is not "' + inherit_interface + 'Prototype"');
+                                    '.prototype is not "' + inherit_interface + '"');
             }
         }
 
         // "The class string of an interface prototype object is the
-        // concatenation of the interface’s qualified identifier and the string
-        // “Prototype”."
+        // interface’s qualified name."
 
-        // Skip these tests for now due to a specification issue about
-        // prototype name.
-        // https://www.w3.org/Bugs/Public/show_bug.cgi?id=28244
+        // "If an object has a class string classString, then the object must,
+        // at the time it is created, have a property whose name is the
+        // @@toStringTag symbol with PropertyDescriptor{[[Writable]]: false,
+        // [[Enumerable]]: false, [[Configurable]]: true, [[Value]]: classString}."
 
-        // assert_class_string(this.get_interface_object().prototype, this.get_qualified_name() + "Prototype",
-        //                     "class string of " + this.name + ".prototype");
+        assert_class_string(this.get_interface_object().prototype, this.get_qualified_name(),
```

Comment:
`assert_class_string` uses `{}.toString.call(object)` and thus more lax and doesn't check everything from the spec quoted above.

How about avoiding `assert_class_string` and instead minting an internal helper using `Object.getOwnPropertyDescriptor(object, Symbol.toStringTag)` and checking the returned descriptor, more like this?

https://github.com/web-platform-tests/wpt/blob/b4c50ee5b945f7c58362f484073b3fa2a3c3a4ef/resources/idlharness.js#L1606-L1612